File: hw4/qb.py
import sys
import math

number = 1
output = open("qbOut.txt", "w")

while True:

    line = sys.stdin.readline()
    if len(line) == 0: output.close()
    nums = line.split(" ")

    completion = int(nums[0])
    attempt = int(nums[1])
    yards = int(nums[2])
    inter = int(nums[3])
    td = int(nums[4])

    if(attempt == 0):
        break

    completion_p = ((completion / attempt) - 0.3) * 5
    td_p = (td/attempt) * 20
    yardPerAttempt = (yards/attempt - 3) * 0.25
    interception = 2.375 - 100 * (inter / ( 4 * attempt ))


    if(completion_p > 2.375):
        completion_p = 2.375
    if(completion_p < 0):
        completion_p = 0

    if(td_p > 2.375):
        td_p = 2.375
    if(td_p < 0):
        td_p = 0
    
    if(yardPerAttempt > 2.375):
        yardPerAttempt = 2.375
    if(yardPerAttempt < 0):
        yardPerAttempt = 0
    
    if(interception > 2.375):
        interception = 2.375
    if(interception < 0):
        interception = 0

    qb = 100 * (completion_p + td_p + yardPerAttempt + interception) / 6

    deci = str(qb - int(qb))

    if(int(deci[2]) >= 5):
        qb = math.ceil(qb)
    else:
        qb = math.floor(qb)

    output.write("QB #%d: %d\n" %(number, qb))
    number += 1

File: hw4/qb.cs
using System;

class qb{
    static void Main(){
        string line;
        int number = 1;

        while((line = Console.ReadLine()) != null && line != ""){
            if(line.Length < 3){
                line = Console.ReadLine();
            }
            string[] nums = line.Split(" ");

            int completion = Int32.Parse(nums[0]);
            int attempt = Int32.Parse(nums[1]);
            int yards = Int32.Parse(nums[2]);
            int inter = Int32.Parse(nums[3]);
            int td = Int32.Parse(nums[4]);
            
            if(attempt == 0){
                break;
            }

            double completion_p = ((completion / attempt) - 0.3) * 5;
            double td_p = (td/attempt) * 20;
            double yardPerAttempt = (yards/attempt - 3) * 0.25;
            double interception = 2.375 - (100 * (inter / ( 4 * attempt )));

            if(completion_p > 2.375){
                completion_p = 2.375;
            }else if(completion_p < 0){
                completion_p = 0;
            }

            if(td_p > 2.375){
                td_p = 2.375;
            }else if(td_p < 0){
                td_p = 0;
            }

            if(yardPerAttempt > 2.375){
                yardPerAttempt = 2.375;
            }else if(yardPerAttempt < 0){
                yardPerAttempt = 0;
            }

            if(interception > 2.375){
                interception = 2.375;
            }else if(interception < 0){
                interception = 0;
            }

            double qb = (100 * (completion_p + td_p + yardPerAttempt + interception)) / 6;

            string deci = (qb - (int)qb).ToString();
        
            if((int)deci[2] >= 5){
                qb = Math.Ceiling(qb);
            }else{
                qb = Math.Floor(qb);
            }

            Console.WriteLine("QB #" + number + ": " + qb);
            number += 1;
        
            }
        }
}